src/rules/netlify-role-management.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @title Netlify Role Management
+ * @overview Adds a default role if the user doesn't have any yet and attaches roles to the ID Token.
+ * @gallery true
+ * @category marketplace
+ *
+ * **Optional configuration:**
+ *
+ *    - `DEFAULT_ROLE_NAME` - name of the default role to be given to a user
+ *    - `DEFAULT_ROLE_ID` - id of the role to be given to a user
+ *    - `CUSTOM_CLAIMS_NAMESPACE` - namespace for adding custom claims to ID Token
+ */
+
+async function netlifyRoleManagement(user, context, callback) {
+  const ManagementClient = require('auth0@2.27.1').ManagementClient;
+
+  const namespace =
+    configuration.CUSTOM_CLAIMS_NAMESPACE || 'https://netlify-integration.com';
+  const assignedRoles = (context.authorization || {}).roles;
```

Comment:
```suggestion
  const assignedRoles = (context.authorization || {}).roles || [];
```

☝️ was not handling the case where the user had no roles.